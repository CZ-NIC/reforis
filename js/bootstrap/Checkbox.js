/*
 * Copyright (C) 2019 CZ.NIC z.s.p.o. (http://www.nic.cz/)
 *
 * This is free software, licensed under the GNU General Public License v3.
 * See /LICENSE for more information.
 */

import React from 'react';
import {LABEL_SIZE} from "./constants";

export default function CheckBox({name, id, label, onChange, checked = false}) {
    return <div className="form-group row">
        <div className={"form-label col-sm-" + LABEL_SIZE}>
            <label className="form-label" htmlFor={id}>{label}</label>
        </div>
        <div className={"form-check"}>
            <input
                type="checkbox"
                name={name}
                id={id}
                onChange={onChange}
                defaultChecked={checked}
            />
        </div>
    </div>;
}
