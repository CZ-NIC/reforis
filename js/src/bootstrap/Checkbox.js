/*
 * Copyright (C) 2019 CZ.NIC z.s.p.o. (http://www.nic.cz/)
 *
 * This is free software, licensed under the GNU General Public License v3.
 * See /LICENSE for more information.
 */

import React from 'react';
import PropTypes from 'prop-types';
import {useUID} from 'react-uid';

import {LABEL_SIZE, FIELD_SIZE} from './constants';

CheckBox.propTypes = {
    label: PropTypes.string.isRequired,
    helpText: PropTypes.string,
};

export default function CheckBox({label, helpText, ...props}) {
    const uid = useUID();
    return <div className='form-group row'>
        <div className={'form-label col-sm-' + LABEL_SIZE}>
            <label className='form-label' htmlFor={uid}>{label}</label>
        </div>
        <div className={'form-check col-sm-' + FIELD_SIZE}>
            <input
                type='checkbox'
                id={uid}
                {...props}
            />
            <small className="form-text text-muted">{helpText}</small>
        </div>
    </div>;
}