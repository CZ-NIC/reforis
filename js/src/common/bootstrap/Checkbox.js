/*
 * Copyright (C) 2019 CZ.NIC z.s.p.o. (http://www.nic.cz/)
 *
 * This is free software, licensed under the GNU General Public License v3.
 * See /LICENSE for more information.
 */

import React from 'react';
import PropTypes from 'prop-types';
import {useUID} from 'react-uid/dist/es5/index';

import {formFieldsSize} from './constants';

CheckBox.propTypes = {
    /** Label message */
    label: PropTypes.string.isRequired,
    /** Help text message */
    helpText: PropTypes.string,
    /** Apply default size (full-width) */
    useDefaultSize: PropTypes.bool,
    /** Control if checkbox is clickable */
    disabled: PropTypes.bool
};

CheckBox.defaultProps = {
    useDefaultSize: true,
    disabled: false
};

export default function CheckBox({label, helpText, useDefaultSize, disabled, ...props}) {
    const uid = useUID();
    return <div className={useDefaultSize ? formFieldsSize : ""} style={{marginBottom: '1rem'}}>
        <div className='custom-control custom-checkbox' style={{marginBottom: '0'}}>
            <input
                className='custom-control-input'
                type='checkbox'
                id={uid}
                disabled={disabled}

                {...props}
            />
            <label className='custom-control-label' htmlFor={uid} style={helpText ? {marginBottom: '0'} : null}>{label}</label>
        </div>
        {helpText ? <small className="form-text text-muted">{helpText}</small> : null}
    </div>
}
