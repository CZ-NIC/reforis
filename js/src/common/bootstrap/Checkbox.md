Checkbox with label Bootstrap component with predefined sizes and structure for using in foris forms.

All additional `props` are passed to the `<input type="checkbox">` HTML component.


```js
import {useState} from 'react';
const [value, setValue] = useState(false);

<CheckBox
    value={value}
    label="Some label" 
    helpText="Read the small text!"
    onChange={value => setValue(value)}
/>
```
