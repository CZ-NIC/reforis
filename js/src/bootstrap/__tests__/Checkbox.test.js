/*
 * Copyright (C) 2019 CZ.NIC z.s.p.o. (http://www.nic.cz/)
 *
 * This is free software, licensed under the GNU General Public License v3.
 * See /LICENSE for more information.
 */

import renderer from 'react-test-renderer';
import React from 'react';
import Checkbox from '../Checkbox'

describe('<Checkbox/>', () => {
    it('Render checkbox', () => {
        const tree = renderer
            .create(
                <Checkbox
                    label="Test label"
                    checked
                    helpText="Some help text"
                />
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('Render uncheked checkbox', () => {
        const tree = renderer
            .create(
                <Checkbox
                    label="Test label"
                    helpText="Some help text"
                />
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});