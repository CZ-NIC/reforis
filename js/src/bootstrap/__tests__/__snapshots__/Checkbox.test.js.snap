// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Checkbox/> Render checkbox 1`] = `
<div
  class="form-group row"
>
  <div
    class="form-label col-sm-3"
  >
    <label
      class="form-label"
      for="1"
    >
      Test label
    </label>
  </div>
  <div
    class="form-check col-sm-8"
  >
    <input
      checked=""
      id="1"
      type="checkbox"
    />
    <small
      class="form-text text-muted"
    >
      Some help text
    </small>
  </div>
</div>
`;

exports[`<Checkbox/> Render uncheked checkbox 1`] = `
<div
  class="form-group row"
>
  <div
    class="form-label col-sm-3"
  >
    <label
      class="form-label"
      for="2"
    >
      Test label
    </label>
  </div>
  <div
    class="form-check col-sm-8"
  >
    <input
      id="2"
      type="checkbox"
    />
    <small
      class="form-text text-muted"
    >
      Some help text
    </small>
  </div>
</div>
`;
